__tests__/app.test.js

```diff
@@ -113,3 +113,91 @@ describe("CORE: GET /api/articles", () => {
       });
   });
 });
+
+describe("CORE: GET /api/articles/:article_id/comments", () => {
+  test("GET: 200, should get all comments for an article", () => {
+    return request(app)
+      .get("/api/articles/1/comments")
+      .expect(200)
+      .then((response) => {
+        const { comments } = response.body;
+        comments.forEach((comment) => {
+          expect(comment.article_id).toBe(1);
+        });
+      });
+  });
+  test("Comments should be served with the most recent comments first. ", () => {
+    return request(app)
+      .get("/api/articles/1/comments")
+      .expect(200)
+      .then((response) => {
+        const { comments } = response.body;
+        expect(comments).toBeSortedBy("created_at", { descending: true });
+      });
+  });
+  test("GET:404 responds with an appropriate status and error message when provided a valid but non existing article id", () => {
+    return request(app)
+      .get("/api/articles/999/comments")
+      .expect(404)
+      .then((response) => {
+        expect(response.body.msg).toBe("Not found");
+      });
+  })
+  //test for valid input
+});
+describe("CORE: POST /api/articles/:article_id/comments", () => {
+  test("POST: 201, should insert a new comment by the article Id", () => {
+    const newComment = {
+      username: "rogersop",
+      body: "This article needs some more comments",
+    };
+    return request(app)
+      .post("/api/articles/5/comments")
+      .send(newComment)
+      .expect(201)
+      .then((response) => {
+        const { comment } = response.body;
+        expect(comment.article_id).toBe(5);
+        expect(comment.author).toBe("rogersop");
+      });
+  })
+  test('GET:404 responds with an appropriate status and error message when provided a valid but non existing article id', () => {
+    const newComment = {
+      username: "rogersop",
+      body: "This article needs some more comments",
+    };
+    return request(app)
+    .post("/api/articles/999/comments")
+    .send(newComment)
+    .expect(404)
+    .then((response) => {
+      expect(response.body.msg).toBe("Not found");
+    });
+  });
+  test('GET:404 responds with an appropriate status and error message when provided a valid but non existing article id', () => {
+    const newComment = {
+      username: "rogersop",
+      body: "This article needs some more comments",
+    };
+    return request(app)
+    .post("/api/articles/999/comments")
+    .send(newComment)
+    .expect(404)
+    .then((response) => {
+      expect(response.body.msg).toBe("Not found");
+    });
+  })
+  test('GET:404 responds with an appropriate status and error message when provided a non existing endpoint', () => {
+    const newComment = {
+      username: "rogersop",
+      body: "This article needs some more comments",
+    };
+    return request(app)
+    .post("/api/articles/5/non-existing-endpoint")
```

Comment:
This should be under a different describe block as it falls under a different end point. 
However, there are a few tests we are missing for POST
Status 201, ignores unnecessary properties
Status 400, invalid ID, e.g. string of "not-an-id"
Status 400, missing required field(s), e.g. no username or body properties
Status 404, username does not exist